custom_components/tahoma/sensor.py

```diff
@@ -393,12 +392,9 @@ def state(self):
 class OverkizHomeKitSetupCodeSensor(OverkizEntity, SensorEntity):
     """Representation of an Overkiz HomeKit Setup Code."""
 
-    def __init__(self, device_url: str, coordinator: OverkizDataUpdateCoordinator):
-        """Initialize the device."""
-        super().__init__(device_url, coordinator)
-        self._attr_name = "HomeKit Setup Code"
-        self._attr_icon = "mdi:shield-home"
-        self._attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_name = "HomeKit Setup Code"
```

Comment:
This won't work, we need to set this in init, otherwise it will still be set to the label.

Reply:
By the way, I broke HomeKit with ignored devices apparently. See my core PR, we need to move the if statement which sets up the entities.

Reply:
> This won't work, we need to set this in init, otherwise it will still be set to the label.

Ah yes you are right. I’m wondering if we cannot fix this.

Reply:
I don't think so, since we set the label in init.